File: interface/src/main/java/com/kgc/pinyougou/service/SpecificationService.java
package com.kgc.pinyougou.service;

import com.kgc.pinyougou.domain.Specification;
import com.kgc.pinyougou.utils.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface SpecificationService {

	List<Map> findAllSpecification();

	/**
	 * @Author Admin
	 * @description:  //TODO
	 * @param:
	 * @param pageNum   分页页数
	 * @param pageSize  分页记录数
	 * @return: com.kgc.pinyougou.utils.PageResult
	 * @time: 2020/4/25 22:11
	 */
	PageResult findPageSpecification(Integer pageNum, Integer pageSize, Specification specification);

	Integer insertSpecification(Specification specification);

	Specification findOne(Long id);

	Integer updateSpecification(Specification specification);

	Integer deleteSpecification(Long [] ids);
}

File: pinyougou-manager-web/src/main/resources/static/js/service/specificationService.js
//服务层
app.service('specificationService',function($http){
    this.findAllSpecification=function () {
        return $http.get("../specification/findAllSpecification")
    }

    this.findPageSpecification=function (page,size,searchEntity) {
        return $http.post("../specification/findPageSpecification?page="+page+"&size="+size,searchEntity)
    }

    this.findOne=function (id) {
        return $http.get("../specification/findOne?id="+id)
    }

    this.del=function (ids) {
        return $http.post("../specification/delSpecification",ids)
    }

    this.add=function (searchEntity) {
        return $http.post("../specification/addSpecification",searchEntity)
    }

    this.update=function (searchEntity) {
        return $http.post("../specification/updateSpecification",searchEntity)
    }
});
